src/services/multiAssetSupply.ts

```diff
@@ -0,0 +1,40 @@
+import { Pool } from "pg";
+import { Request, Response } from "express";
+
+interface Asset {
+  name: string
+  policy: string
+}
+
+export const handleGetMultiAssetSupply = (pool: Pool) => async (req: Request, res:Response): Promise<void> => {
+  if (!req.body || !req.body.assets) throw new Error("missing assets on request body");
+  if (!Array.isArray(req.body.assets)) throw new Error("assets should be an array");
+  if (req.body.assets.length === 0) throw new Error("assets should not be empty");
+  if (req.body.assets.find((a: any) => !a.policy)) throw new Error("all assets on body should have a name and a policy");
+
+  const assets: Asset[] = req.body.assets;
+
+  const ret = await Promise.all(assets.map(async (asset) => {
+    const supply = await getMultiAssetSupply(pool, asset);
+
+    const policyAndName = `${asset.policy}.${asset.name}`;
+    
+    return supply;
+  }));
+
+  res.send(ret);
+};
+
+const getMultiAssetSupply = async (pool: Pool, asset: Asset): Promise<number> => {
+  const query = `
+    select sum(quantity) as supply
+    from ma_tx_mint mint
+    where
+      mint.name = ($1)::bytea and encode(mint.policy, 'hex') = ($2)::varchar`;
```

Comment:
@ozgrakkurt , this is for the current version of db-sync in production, correct?

Reply:
Yes, this is for the production/master version. Not the beta